cmem/encoder.go

```diff
@@ -54,6 +54,11 @@ func (enc *Encoder) Encode(data interface{}) error {
 	}
 
 	rv := reflect.Indirect(reflect.ValueOf(data))
+	if !rv.IsValid() {
+		return fmt.Errorf("cmem: reflect.ValueOf returned invalid value for "+
```

Comment:
I wouldn't split that string. ok, it runs a bit long but who cares.
alternatively:
```go
return fmt.Errorf(
    "cmem: reflect.ValueOf returned invalid value for type %T",
    data,
)
```

Reply:
I care... Hehe :P But I can change it to your suggestion.

Reply:
Yeah, please don't do this. We're not using Fortran and we're not on punch cards.